sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClientBuilder.java

```diff
@@ -114,10 +104,26 @@ public ChatThreadClientBuilder addPolicy(HttpPipelinePolicy pipelinePolicy) {
         return this;
     }
 
+    /**
+     * Sets the {@link RetryPolicy} that will attempt to retry failed requests, if applicable.
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @throws NullPointerException if retryPolicy is null.
+     * @return The updated {@link ChatThreadClientBuilder} object.
+     */
+    public ChatThreadClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        if (retryPolicy == null) {
```

Comment:
`retryPolicy` can be `null` if we have a default value to set it to when not present, which we do below in `applyRequiredPolicies()`

Reply:
Yes. It can be null. If user does not call this function, it would use the default retry policy. Here we only want to check when user calls retryPolicy(RetryPolicy retryPolicy), he should provide a non-null retryPolicy.